packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js

```diff
@@ -139,10 +151,15 @@ const HomePage = () => {
     });
   };
 
+  const handleDeleteMedias = async () => {
+    await Promise.all(dataToDelete.map(id => deleteMedia(id)));
```

Comment:
You might need to re request your data after this and also make sure that the component is still mounted before setting your state.